modules/v2/main.tf

```diff
@@ -32,6 +32,31 @@ locals {
     email  = google_service_account.sa[0].email,
     member = google_service_account.sa[0].member
   } : {}
+
+  ingress_container = try(
+    [for container in var.containers : container if length(try(container.ports, {})) > 0][0],
+    null
+  )
+  prometheus_sidecar_container = [{
+    container_name  = "collector"
+    container_image = "us-docker.pkg.dev/cloud-ops-agents-artifacts/cloud-run-gmp-sidecar/cloud-run-gmp-sidecar:1.1.1"
+    # Set default values for the sidecar container
+    ports                = {}
+    working_dir          = null
+    depends_on_container = [local.ingress_container.container_name]
```

Comment:
Is there a reason to take dependency on only one container with port defined! Even without port definition, the container image itself can run on a default port defined in the Dockerfile of the container image.
IMO we should take dependency on all the user created containers!

Reply:
I set up this `ingress_container` based on two findings:
1. It seems like only one container can expose port in a cloud run instance. I can't find an official document on that at the moment but I got errors when trying to have two containers both expose ports. https://stackoverflow.com/q/61355416 sort of support that.
2. I don't think there will be a case that port is not defined, because we will use `8080` as the default port if that's not set up explicitly. For example, the v2_with_gmp example here we are not setting up the port but when the `locals` are being interpreted the container already has an assigned default port.

For the reasons above I use this `ingress_container` to make sure the Prometheus sidecar monitors the metrics from the main process. More context can be found in https://cloud.google.com/run/docs/container-contract#port.